### PR TITLE
Bump opentelemetry to v0.15.0

### PR DESCRIPTION
**Pull request by Vemmy124**
## Which problem is this PR solving?
Update of dependency

## Short description of the changes
The only real thing is that any `pdata.Traces` component is non-nullable now, so `IsNil()` and `InitEmpty()` became deprecated.

No changes applied to logic. The config for jaegerreceiver was updated in opentelemetry (initiated and implemented by me actually), but it is orthogonal to current jaeger code.

### CODE REVIEW

File: cmd/opentelemetry/app/exporter/elasticsearchexporter/esmodeltranslator/modeltranslator.go [hunk 1]
```diff
@@ -248,7 +248,7 @@ func traceIDToString(high, low uint64) string {
 }
 
 func (c *Translator) process(resource pdata.Resource) *dbmodel.Process {
-	if resource.IsNil() || resource.Attributes().Len() == 0 {
+	if resource.Attributes().Len() == 0 {
```
Comment (joe-elliott) [inline cmd/opentelemetry/app/exporter/elasticsearchexporter/esmodeltranslator/modeltranslator.go:251]: It seems that `IsNil()` was removed and the resource is guaranteed to never be nil?

I find these generate proto wrappers very confusing.  Do you have any insight into why this was removed?

File: cmd/opentelemetry/app/defaultconfig/default_config.go [hunk 2]
```diff
@@ -206,13 +206,15 @@ func createExporters(component ComponentType, storageTypes string, factories com
 
 func enableAgentUDPEndpoints(jaeger *jaegerreceiver.Config) {
 	if jaeger.ThriftCompact == nil {
-		jaeger.ThriftCompact = &confignet.TCPAddr{
+		jaeger.ThriftCompact = &jaegerreceiver.ProtocolUDP{
```
Comment (joe-elliott) [inline cmd/opentelemetry/app/defaultconfig/default_config.go:209]: Nice.  This will allow us to support max udp packet size in the Jaeger otel collectors.

**Review by joe-elliott — DISMISSED** [review]
This looks good to me.  I do have one question about these proto wrappers.  Any insight you have would be appreciated.

File: cmd/opentelemetry/app/exporter/elasticsearchexporter/esmodeltranslator/modeltranslator.go (hunk 1 above)
Comment (Vemmy124) [reply cmd/opentelemetry/app/exporter/elasticsearchexporter/esmodeltranslator/modeltranslator.go:251]: Well, that went from my issue about panic in marshalling https://github.com/open-telemetry/opentelemetry-collector/issues/1985. Then gradually all proto things became non-nullable. I believe, that using pointers was a mistake from the very beginning

File: cmd/opentelemetry/app/defaultconfig/default_config.go (hunk 2 above)
Comment (Vemmy124) [reply cmd/opentelemetry/app/defaultconfig/default_config.go:209]: Yeah, it is already configurable in otel receiver config

File: cmd/opentelemetry/app/exporter/elasticsearchexporter/esmodeltranslator/modeltranslator.go (hunk 1 above)
Comment (Vemmy124) [reply cmd/opentelemetry/app/exporter/elasticsearchexporter/esmodeltranslator/modeltranslator.go:251]: Yes, you're right, resource now cannot be nil

File: cmd/opentelemetry/app/defaultconfig/default_config.go (hunk 2 above)
Comment (Vemmy124) [reply cmd/opentelemetry/app/defaultconfig/default_config.go:209]: Alongwith socket buffer size, number of workers and queue size

**Comment by Vemmy124** [comment]
This is a flaky test I suppose

**Review by joe-elliott — APPROVED** [review]
Looks good.

**Review by yurishkuro — APPROVED** [review]

**Pull request merged**